src/frontegg-mock/src/lib.rs

```diff
@@ -538,7 +553,161 @@ async fn handle_post_user_api_token<'a>(
+    };
+
+    let tenant_api_tokens = context.tenant_api_tokens.lock().unwrap();
+    let tokens: Vec<TenantApiTokenResponse> = tenant_api_tokens
+        .iter()
+        .map(|(api_token, config)| TenantApiTokenResponse {
+            client_id: api_token.client_id,
+            description: "".to_string(),
+            secret: api_token.secret,
+            created_by_user_id: Uuid::nil(),
+            metadata: config
+                .metadata
+                .as_ref()
+                .and_then(|m| m.user.clone().map(serde_json::Value::String)),
+            created_at: Utc::now(),
+            role_ids: config.roles.clone(),
+        })
+        .collect();
+
+    Ok(Json(tokens))
+}
+
+// https://docs.frontegg.com/reference/tenantapitokensv1controller_createtenantapitoken
+async fn handle_create_tenant_api_token(
+    State(context): State<Arc<Context>>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+    Json(request): Json<CreateTenantApiTokenRequest>,
+) -> Result<(StatusCode, Json<TenantApiTokenResponse>), StatusCode> {
+    let claims = match decode_access_token(&context, authorization.token()) {
+        Ok(TokenData { claims, .. }) => claims,
+        Err(_) => return Err(StatusCode::UNAUTHORIZED),
+    };
+
+    let new_token = ApiToken {
+        client_id: Uuid::new_v4(),
+        secret: Uuid::new_v4(),
+    };
+
+    let config = TenantApiTokenConfig {
+        tenant_id: claims.tenant_id,
+        metadata: request.metadata.as_ref().and_then(|m| {
+            if let serde_json::Value::String(s) = m {
+                Some(ClaimMetadata {
+                    user: Some(s.clone()),
+                })
+            } else {
+                None
+            }
+        }),
+        roles: request.role_ids.clone(),
+    };
+
+    let mut tenant_api_tokens = context.tenant_api_tokens.lock().unwrap();
+    tenant_api_tokens.insert(new_token.clone(), config);
+
+    let response = TenantApiTokenResponse {
+        client_id: new_token.client_id,
+        description: request.description,
+        secret: new_token.secret,
+        created_by_user_id: claims.sub,
+        metadata: request.metadata,
+        created_at: Utc::now(),
+        role_ids: request.role_ids,
+    };
+
+    Ok((StatusCode::CREATED, Json(response)))
+}
+
+// https://docs.frontegg.com/reference/tenantapitokensv1controller_deletetenantapitoken
+async fn handle_delete_tenant_api_token(
+    State(context): State<Arc<Context>>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+    Path(token_id): Path<Uuid>,
+) -> StatusCode {
+    let _claims = match decode_access_token(&context, authorization.token()) {
+        Ok(TokenData { claims, .. }) => claims,
+        Err(_) => return StatusCode::UNAUTHORIZED,
+    };
+
+    let mut tenant_api_tokens = context.tenant_api_tokens.lock().unwrap();
+    if tenant_api_tokens
+        .remove(&ApiToken {
+            client_id: token_id,
+            secret: Uuid::nil(),
```

Comment:
This won't match your actual api token, so this will always be not found, right?